File: hexbound_backend/Hexbound.API/Models/TokenStats.cs
namespace Hexbound.API.Models;

/// <summary>
/// Represents a token's combat statistics
/// </summary>
public class TokenStats
{
    public string TokenId { get; set; } = string.Empty;
    public string Name { get; set; } = "Unknown";
    public int MaxHp { get; set; } = 10;
    public int CurrentHp { get; set; } = 10;
    public int ArmorClass { get; set; } = 10;
    public int Speed { get; set; } = 6; // Hexes per turn
    public int InitiativeModifier { get; set; } = 0;
    public List<string> Conditions { get; set; } = new();
    
    // Position (synced with token position)
    public int Q { get; set; }
    public int R { get; set; }
    
    // Helper properties
    public bool IsAlive => CurrentHp > 0;
    public double HpPercentage => MaxHp > 0 ? (double)CurrentHp / MaxHp : 0;
}

File: hexbound_client/lib/game/models/token_stats.dart
/// Token Combat Statistics Model
class TokenStats {
  final String tokenId;
  final String name;
  final int maxHp;
  int currentHp;
  final int armorClass;
  final int speed;
  final int initiativeModifier;
  List<String> conditions;
  int q;
  int r;

  TokenStats({
    required this.tokenId,
    this.name = "Unknown",
    this.maxHp = 10,
    int? currentHp,
    this.armorClass = 10,
    this.speed = 6,
    this.initiativeModifier = 0,
    List<String>? conditions,
    this.q = 0,
    this.r = 0,
  })  : currentHp = currentHp ?? maxHp,
        conditions = conditions ?? [];

  bool get isAlive => currentHp > 0;
  double get hpPercentage => maxHp > 0 ? currentHp.toDouble() / maxHp.toDouble() : 0.0;

  factory TokenStats.fromJson(Map<String, dynamic> json) => TokenStats(
        tokenId: json['tokenId'] ?? '',
        name: json['name'] ?? 'Unknown',
        maxHp: json['maxHp'] ?? 10,
        currentHp: json['currentHp'],
        armorClass: json['armorClass'] ?? 10,
        speed: json['speed'] ?? 6,
        initiativeModifier: json['initiativeModifier'] ?? 0,
        conditions: List<String>.from(json['conditions'] ?? []),
        q: json['q'] ?? 0,
        r: json['r'] ?? 0,
      );

  Map<String, dynamic> toJson() => {
        'tokenId': tokenId,
        'name': name,
        'maxHp': maxHp,
        'currentHp': currentHp,
        'armorClass': armorClass,
        'speed': speed,
        'initiativeModifier': initiativeModifier,
        'conditions': conditions,
        'q': q,
        'r': r,
      };
}
